File: frontend/src/api/summary.ts
import { api } from './client'

export interface HourlyPoint {
  hour: string
  units: number
}

export interface ResinPoint {
  resin: string
  units: number
  litres: number
}

export interface CartridgePoint {
  cartridge_type: string
  units: number
  litres: number
}

export interface ShiftSummary {
  has_logs_today: boolean
  units: number
  litres: number
  scrap: number
  yield_pct: number
  logs_submitted: number
  has_output_logs: boolean
  hourly_timeline: HourlyPoint[]
  by_resin: ResinPoint[]
  by_cartridge: CartridgePoint[]
}

export interface MonthlyRecap {
  has_data: boolean
  month_label: string
  units: number
  best_day_ordinal: string | null
  best_day_units: number
  mismatches: number
}

export const summaryApi = {
  today: (asOperator?: string) =>
    api.get<ShiftSummary>(asOperator ? `/summary/today?${new URLSearchParams({ as_operator: asOperator })}` : '/summary/today'),
  monthly: () => api.get<MonthlyRecap>('/summary/monthly'),
}

File: api/schemas/summary.py
from pydantic import BaseModel


class HourlyPoint(BaseModel):
    hour: str  # ISO 8601, floored to the hour
    units: int


class ResinPoint(BaseModel):
    resin: str
    units: int
    litres: float = 0.0


class CartridgePoint(BaseModel):
    cartridge_type: str
    units: int
    litres: float = 0.0


class ShiftSummaryOut(BaseModel):
    has_logs_today: bool
    units: int
    litres: float = 0.0
    scrap: int
    yield_pct: float
    logs_submitted: int
    has_output_logs: bool  # False when today's only logs are e.g. downtime/audit
    hourly_timeline: list[HourlyPoint] = []
    by_resin: list[ResinPoint] = []
    by_cartridge: list[CartridgePoint] = []


class MonthlyRecapOut(BaseModel):
    has_data: bool
    month_label: str  # e.g. "September"
    units: int
    best_day_ordinal: str | None = None  # e.g. "12th", formatted server-side
    best_day_units: int = 0
    mismatches: int = 0
